cli/upgrade.rs

```diff
@@ -239,14 +260,31 @@ fn test_find_version() {
 #[test]
 fn test_compose_url_to_exec() {
   let v = semver_parse("0.0.1").unwrap();
-  let url = compose_url_to_exec(&v).unwrap();
-  #[cfg(windows)]
-  assert_eq!(url.as_str(), "https://github.com/denoland/deno/releases/download/v0.0.1/deno-x86_64-pc-windows-msvc.zip");
-  #[cfg(target_os = "macos")]
+  let r = compose_url_to_exec(&v);
+  #[cfg(all(target_os = "windows", target_arch = "x86_64"))]
   assert_eq!(
-    url.as_str(),
+    r.unwrap().as_str(),
+    "https://github.com/denoland/deno/releases/download/v0.0.1/deno-x86_64-pc-windows-msvc.zip"
+  );
+  #[cfg(all(target_os = "macos", target_arch = "x86_64"))]
+  assert_eq!(
+    r.unwrap().as_str(),
     "https://github.com/denoland/deno/releases/download/v0.0.1/deno-x86_64-apple-darwin.zip"
   );
-  #[cfg(target_os = "linux")]
-  assert_eq!(url.as_str(), "https://github.com/denoland/deno/releases/download/v0.0.1/deno-x86_64-unknown-linux-gnu.zip");
+  #[cfg(all(target_os = "linux", target_arch = "x86_64"))]
+  assert_eq!(
+    r.unwrap().as_str(),
+    "https://github.com/denoland/deno/releases/download/v0.0.1/deno-x86_64-unknown-linux-gnu.zip"
+  );
+  #[cfg(any(
```

Comment:
Need to think through this logic. Maybe the following would be better (assuming `not(all(xyz))` works):

```rust
#[cfg(all(
	not(all(target_os = "windows", target_arch = "x86_64"))
	not(all(target_os = "macos", target_arch = "x86_64"))
	not(all(target_os = "linux", target_arch = "x86_64"))
))]
```

Reply:
This was dropped